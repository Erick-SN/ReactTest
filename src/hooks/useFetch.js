import { useState, useEffect } from 'react';
import { getGifs } from '../helpers/getGifts';

export const useFetch = (category) => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  });
  useEffect(() => {
    getGifs(category).then((data) => {
      setState({
        data: data,
        loading: false,
      });
    });
  }, [category]);
  return state;
};
