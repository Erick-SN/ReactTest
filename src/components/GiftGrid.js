import GiftItem from './GiftItem';
import PropTypes from 'prop-types';
import { useFetch } from '../hooks/useFetch';
const GiftGrid = ({ category }) => {
  const { data, loading } = useFetch(category);
  return (
    <>
      <h3>{category}</h3>
      {loading && <p>Loading...</p>}
      <div className="card-gift">
        {data.map((img) => (
          <GiftItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};
GiftGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
export default GiftGrid;
