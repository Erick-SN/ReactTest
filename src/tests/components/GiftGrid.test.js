import { useFetch } from '../../hooks/useFetch';
import { shallow } from 'enzyme';
import GiftGrid from '../../components/GiftGrid';
jest.mock('../../hooks/useFetch');
describe('Pruebas en <GiftGrid/>', () => {
  let wrapper;
  beforeEach(() => {
    jest.clearAllMocks();
    useFetch.mockReturnValue({
      data: [],
      loading: true,
    });
    wrapper = shallow(<GiftGrid category="" />);
  });
  test('Prueba de renderizado correcto', () => {
    // useFetch.mockReturnValue({
    //   data: [],
    //   loading: true,
    // });
    // const wrapper = shallow(<GiftGrid category="Goku" />);
    expect(wrapper).toMatchSnapshot();
  });
  test('Debe de mostrar items al ejecutar useFetch', () => {
    const gifs = [
      {
        id: 'ABC',
        url: 'https://google.com',
        title: 'abc',
      },
    ];
    useFetch.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GiftGrid category="Goku" />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GiftItem').length).toBe(gifs.length);
  });
});
