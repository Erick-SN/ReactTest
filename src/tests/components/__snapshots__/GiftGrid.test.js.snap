// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftGrid/> Debe de mostrar items al ejecutar useFetch 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <div
    className="card-gift"
  >
    <GiftItem
      id="ABC"
      key="ABC"
      title="abc"
      url="https://google.com"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GiftGrid/> Prueba de renderizado correcto 1`] = `
<Fragment>
  <h3 />
  <p>
    Loading...
  </p>
  <div
    className="card-gift"
  />
</Fragment>
`;
